platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/DtoConverter.java

```diff
@@ -158,170 +126,109 @@ public static ItemReference toItemReference(FolderEntry folder,
-            dto.withType(BaseProjectType.ID);
-        }
-
-        if (config != null) {
-            dto.withDescription(config.getDescription());
-//            dto.withRecipe(config.getSource().getLocation());
-            String typeId = config.getType();
-            dto.withType(typeId).withTypeName(ptRegistry.getProjectType(typeId).getDisplayName()).withMixins(config.getMixins());
-
-            dto.withAttributes(config.getAttributes());
-
-            List<ModuleConfigDto> modules = config.getModules().stream().map(DtoConverter::toModuleConfigDto).collect(Collectors.toList());
-
-            dto.withModules(modules);
-        }
-
-        final User currentUser = environmentContext.getUser();
-        List<AccessControlEntry> acl = null;
-        try {
-            acl = project.getPermissions();
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
-        if (acl != null) {
-            final List<String> permissions = new LinkedList<>();
-            if (acl.isEmpty()) {
-                // there is no any restriction at all
-                permissions.add("all");
-            } else {
-                for (AccessControlEntry accessControlEntry : acl) {
-                    final Principal principal = accessControlEntry.getPrincipal();
-                    if ((Principal.Type.USER == principal.getType() && currentUser.getId().equals(principal.getName()))
-                        || (Principal.Type.USER == principal.getType() && "any".equals(principal.getName()))
-                        || (Principal.Type.GROUP == principal.getType() && currentUser.isMemberOf(principal.getName()))) {
-
-                        permissions.addAll(accessControlEntry.getPermissions());
-                    }
-                }
-            }
-            dto.withPermissions(permissions);
-        }
-
-        try {
-            dto.withCreationDate(project.getCreationDate());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
-
-        try {
-            dto.withModificationDate(project.getModificationDate());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
-
-        try {
-            dto.withVisibility(project.getVisibility());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
+    /**
+     * The method tries to provide as much as possible information about project.If get error then save information about error
+     * with 'problems' field in ProjectConfigDto.
+     *
+     * @param project
+     *         project from which we need get information
+     * @param serviceUriBuilder
+     *         service for building URI
+     * @return an instance of {@link ProjectConfigDto}
+     * @throws InvalidValueException
```

Comment:
Describe exception javadoc as well. [Doc](https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#javadoc) says that _@param, @return, @throws or @deprecated must not be appeared without of description_